packages/firestore/src/local/memory_query_cache.ts

```diff
@@ -172,7 +217,14 @@ export class MemoryQueryCache implements QueryCache {
     targetId: TargetId
   ): PersistencePromise<void> {
     this.references.removeReferences(keys, targetId);
-    return PersistencePromise.resolve();
+    const referenceDelegate = this.persistence.referenceDelegate;
+    const promises: Array<PersistencePromise<void>> = [];
+    if (referenceDelegate) {
+      keys.forEach(key => {
+        promises.push(referenceDelegate.removeReference(txn, key));
```

Comment:
Same optional comment as above.

Reply:
Again, we don't know if the number of keys is small.